src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -25,44 +26,92 @@ public class ProcessStartInfo
     /// </summary>
     public class Container : IDisposable
     {
-        private string _id;
+        private IntPtr _cs;
         private bool _killOnClose;
-        private bool _killed;
+        private bool _dead;
 
-        internal Container(string id, bool terminateOnClose)
+        private HcsNotificationWatcher _watcher;
+
+        private Container(string id, IntPtr computeSystem, bool terminateOnClose, HcsNotificationWatcher watcher)
         {
-            _id = id;
             _killOnClose = terminateOnClose;
+            _cs = computeSystem;
+            _watcher = watcher;
+        }
+
+        internal static Container Initialize(string id, IntPtr computeSystem, bool terminateOnClose)
+        {
+            var watcher = new HcsNotificationWatcher(
+                computeSystem,
+                HcsFunctions.HcsRegisterComputeSystemCallback,
+                HcsFunctions.HcsUnregisterComputeSystemCallback,
+                new HCS_NOTIFICATIONS[]{
+                    HCS_NOTIFICATIONS.HcsNotificationSystemExited,
+                    HCS_NOTIFICATIONS.HcsNotificationSystemCreateCompleted,
+                    HCS_NOTIFICATIONS.HcsNotificationSystemStartCompleted,
+                    HCS_NOTIFICATIONS.HcsNotificationServiceDisconnect
```

Comment:
Yeah, I think leaving it in the enum but taking it out of the list of watched events makes sense.